src/OrchardCore/OrchardCore.Media.Abstractions/IMediaFileStore.cs

```diff
@@ -13,4 +14,28 @@ public interface IMediaFileStore : IFileStore
     /// <param name="path">The path within the file store.</param>
     /// <returns>A string containing the mapped public URL of the given path.</returns>
     string MapPathToPublicUrl(string path);
+
+    /// <summary>
+    /// Creates a new file in the file store from the contents of an input stream.
+    /// </summary>
+    /// <param name="context">The <see cref="MediaCreatingContext"/> which contains the path of the file to be created.</param>
+    /// <param name="inputStream">The stream whose contents to write to the new file.</param>
+    /// <param name="overwrite"><c>true</c> to overwrite if a file already exists; <c>false</c> to throw an exception if the file already exists.</param>
+    /// <returns>
+    /// A tuple containing:
+    /// <list type="bullet">
+    /// <item>
+    /// <description><c>outputPath</c>: The final path of the created file within the file store.</description>
+    /// </item>
+    /// <item>
+    /// <description><c>outputStream</c>: The resulting stream, which may be the original input stream or a modified stream if any event handlers altered it.</description>
+    /// </item>
+    /// </list>
+    /// </returns>
+    /// <remarks>
+    /// If the specified path contains one or more directories, then those directories are
+    /// created if they do not already exist.
+    /// </remarks>
+    Task<(string outputPath, Stream outputStream)> CreateMediaFileFromStreamAsync(MediaCreatingContext context, Stream inputStream, bool overwrite = false) => Task.FromResult((context.Path, inputStream));
```

Comment:
To change the interface in a backward compatible way, the default implementation must actually work. So you need to call the existing method here. E.g.

```csharp
async Task<(string outputPath, Stream outputStream)> CreateMediaFileFromStreamAsync(MediaCreatingContext context, Stream inputStream, bool overwrite = false)
{
    var outputPath = await CreateFileFromStreamAsync(context.Path, inputStream, overwrite);
    return (outputPath, inputStream);
}
```